Hazel/src/Hazel/Renderer/OrthographicCameraController.cpp

```diff
@@ -13,34 +13,21 @@ namespace Hazel {
 
 	void OrthographicCameraController::OnUpdate(Timestep ts)
 	{
-		if (Input::IsKeyPressed(HZ_KEY_A))
-		{
-			m_CameraPosition.x -= cos(glm::radians(m_CameraRotation)) * m_CameraTranslationSpeed * ts;
-			m_CameraPosition.y -= sin(glm::radians(m_CameraRotation)) * m_CameraTranslationSpeed * ts;
-		}
-		else if (Input::IsKeyPressed(HZ_KEY_D))
-		{
-			m_CameraPosition.x += cos(glm::radians(m_CameraRotation)) * m_CameraTranslationSpeed * ts;
-			m_CameraPosition.y += sin(glm::radians(m_CameraRotation)) * m_CameraTranslationSpeed * ts;
-		}
-
-		if (Input::IsKeyPressed(HZ_KEY_W))
-		{
-			m_CameraPosition.x += -sin(glm::radians(m_CameraRotation)) * m_CameraTranslationSpeed * ts;
-			m_CameraPosition.y += cos(glm::radians(m_CameraRotation)) * m_CameraTranslationSpeed * ts;
-		}
-		else if (Input::IsKeyPressed(HZ_KEY_S))
-		{
-			m_CameraPosition.x -= -sin(glm::radians(m_CameraRotation)) * m_CameraTranslationSpeed * ts;
-			m_CameraPosition.y -= cos(glm::radians(m_CameraRotation)) * m_CameraTranslationSpeed * ts;
-		}
+		float dTranslationSpeed = m_CameraTranslationSpeed * ts;
+		float dRotationSpeed    = m_CameraRotationSpeed * ts;
+		float theta = m_CameraRotation;
+		float dX = cos(glm::radians(theta));
+		float dY = sin(glm::radians(theta));
 
+		if (Input::IsKeyPressed(HZ_KEY_A)) ChangePosition(-(dX * dTranslationSpeed), -(dY * dTranslationSpeed));
+		if (Input::IsKeyPressed(HZ_KEY_D)) ChangePosition(dX * dTranslationSpeed, dY * dTranslationSpeed);
+		if (Input::IsKeyPressed(HZ_KEY_W)) ChangePosition(-dY * dTranslationSpeed, dX * dTranslationSpeed);
+		if (Input::IsKeyPressed(HZ_KEY_S)) ChangePosition(-(-dY * dTranslationSpeed), -(dX * dTranslationSpeed));
```

Comment:
As @Puddlestomper said, I'm not a fan of this implementation:
1. The name `ChangePosition` would imply you would give it an absolute value, not a delta (`movePosition` would be more appropriate for example?)
2. I don't really like the functions `ChangePosition` and `ChangeRotation`, I would rather take a compromise between both implementations: calculate the sine & cosine before and use the `dCos`, `dSin` and `dRot` in the if statements. (the names `dx` `dy` and `dtheta` don't realy read nicely, for example check the W and S keys...)

This is just my opinion... I don't really see any improvements over the current implementation other than maybe some readability...

Reply:
I agree with dropping the methods. This PR is really just for readability, so introducing the new methods isn't really needed.

Reply:
I agree with @LovelySanta that ``Change *`` is not really needed. I just roughly suggested how to implement this. Calculation of sine and cosine, I originally wanted to make it calculated above as one line (or two). But I just write my engine in a slightly different way, and just relying on the lessons of Cherno and I don’t know if this will work correctly in Hazel. I'll see if I can still find better ways to do this.  

This is just my suggestion, and I'm almost 99% sure that Cherno will somehow implement it in its own way. Especially when GameObject's system appears.